boa/src/builtins/array/mod.rs

```diff
@@ -957,6 +957,76 @@ impl Array {
         Ok(Value::from(false))
     }
 
+    /// `Array.prototype.reduce( callbackFn [ , initialValue ] )`
+    ///
+    /// The reduce method traverses left to right starting from the first defined value in the array,
+    /// accumulating a value using a given callback function. It returns the accumulated value.
+    ///  
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-array.prototype.reduce
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce
+    pub(crate) fn reduce(
+        this: &Value,
+        args: &[Value],
+        interpreter: &mut Interpreter,
+    ) -> ResultValue {
+        let this = interpreter.to_object(this)?;
+        let callback = match args.get(0) {
+            Some(value) if value.is_function() => value,
+            _ => {
+                return interpreter.throw_type_error(
+                    "missing callback when calling function Array.prototype.reduce",
+                )
+            }
+        };
+        let initial_value = args.get(1).cloned().unwrap_or_else(Value::undefined);
+        let mut length = i32::from(&this.get_field("length"));
```

Comment:
```suggestion
        let mut length = interpereter.to_length(&this.get_field("length"))?;
```
We should use `.to_length()` method instead of `i32::from` because it is not spec compliant and we are in the process of removing it. :)